File: daily/242.有效的字母异位词.js
/*
 * @lc app=leetcode.cn id=242 lang=javascript
 *
 * [242] 有效的字母异位词
 */

// @lc code=start
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
  if (s.length !== t.length) return false;
  const map = new Map();
  for (let c of s) {
    map.set(c, map.has(c) ? map.get(c) + 1 : 1);
  }
  for (let c of t) {
    if (!map.has(c)) return false;
    const count = map.get(c);
    if (count < 1) return false;
    else map.set(c, count - 1);
  }
  return true;
};
// @lc code=end

console.log(isAnagram('ab', 'a'));

File: daily/242.有效的字母异位词.rs
/*
 * @lc app=leetcode.cn id=242 lang=rust
 *
 * [242] 有效的字母异位词
 */

// @lc code=start
impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        if s.len() != t.len() { return false; }
        let mut arr = vec![0; 26];
        for c in s.chars() {
            arr[c as usize - 97] += 1;
        }
        for c in t.chars() {
            if arr[c as usize - 97] < 1 { return false; }
            else { arr[c as usize - 97] -= 1; }
        }
        return true;
    }
}
// @lc code=end

impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        let (mut s, mut t) = (s.into_bytes(), t.into_bytes());
        s.sort_unstable();
        t.sort_unstable();
        s == t
    }
}

